app/posts/register-trainee-teachers/2022-02-21-autocomplete-improvements.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Improvements to our autocompletes
+description: To add
+date: 2022-02-21
+related:
+  items:
+  - text: Using analytics to understand where our autocompletes could be improved
+    href: /register-trainee-teachers/autocomplete-analytics/
+  - text: Investigating free text responses from Apply
+    href: /register-trainee-teachers/investigating-free-text-responses-from-apply/
+  - text: Autocomplete analytics dashboard (Register team members only)
+    href: https://datastudio.google.com/s/qW6s8xj8O-I
+---
+{% from "screenshots/macro.njk" import appScreenshots with context %}
+{% from "../../../node_modules/govuk-frontend/govuk/components/table/macro.njk" import govukTable with context %}
+
+Over the last year we’ve updated our autocompletes to improve how they work and address usability issues we’ve found. We’ve tried to make these improvements general purpose so that other teams can benefit from them.
+
+For most projects you should be able to ‘drop-in’ our new code. You can add extra configuration with data attributes, or leave as is to get better results sorted in a more useful way.
+
+## Problems we’re trying to overcome
+
+### Filtering available items
+
+The autocomplete takes your search query and looks for an *exact match* in the available entries. This excludes close matches or ones where there’s an extra word or punctuation in between.
+
+For example, these searches will fail:
+
+{{ govukTable({
+  firstCellIsHeader: false,
+  head: [
+    {
+      text: "Search term"
+    },
+    {
+      text: "Should match but gets missed"
+    }
+  ],
+  rows: [
+    [
+      {
+        text: "Bachelor art"
+      },
+      {
+        text: "Bachelor of art"
+      }
+    ],
+    [
+      {
+        text: "dh law"
+      },
+      {
+        text: "D.H. Lawrence Studies"
+      }
+    ],
+    [
+      {
+        text: "Bedfordshire university"
+      },
+      {
+        text: "University of Bedfordshire"
+      }
+    ],
+    [
+      {
+        text: "The University of Bedfordshire"
+      },
+      {
+        text: "University of Bedfordshire"
+      }
+    ]
+  ]
+}) }}
+
+
+For each of these examples, the search term gives us enough information to know what the user meant, but the filtering algorithm is too strict.
+
+### Sorting filtered results
+
+The autocomplete does not sort results. It returns results ordered in the same order as the source data. If the source data was alphabetical, then the results will be too. This means that more likely matches may appear some way down the list.
+
+Although alphabetical may seem natural, users tend to expect that autocompletes will return relevant results first. This is particularly important when there are lots of results - users don’t need to read many to get to the more relevant ones. If what the user has typed exactly matches an option, it would be sensible for that option to be suggested first.
```

Comment:
```suggestion
Although alphabetical may seem natural, users tend to expect that autocompletes will return relevant results first. This is particularly important when there are lots of results — users don’t need to read many to get to the more relevant ones. If what the user has typed exactly matches an option, it would be sensible for that option to be suggested first.
```

Reply:
"users don’t need to read many to get to the more relevant ones" I'm not sure I follow what you mean here.